crates/rnote-ui/src/appwindow/mod.rs

```diff
@@ -492,6 +492,38 @@ impl RnAppWindow {
         self.main_header().main_title().set_subtitle(&subtitle);
     }
 
+    pub(crate) fn refresh_pages(&self, canvas: &RnCanvas) {
+        let layout = canvas.engine_ref().document.layout;
+        match layout {
+            Layout::FixedSize | Layout::ContinuousVertical => {
+                self.main_header().page_spin().set_visible(true);
+                let n_pages = canvas.engine_ref().document.calc_n_pages();
+                self.main_header()
+                    .page_spin()
+                    .set_range(1.0, n_pages as f64);
+
+                // zoom need to be taken into account
+
+                // slight offset : for now, because going to the page will
+                // give the previous page when the go_to_page function is called
+                // hence a cascade of calls are made until the first page is shown
+                // maybe we should set the page number based on the page visible
+                // at the center of the viewport instead
+                let page_number = ((canvas.engine_ref().camera.offset().y + 0.01)
+                    / (canvas.engine_ref().document.format.height()
+                        * canvas.engine_ref().camera.zoom()))
+                .ceil()
+                .max(1.0);
+                self.main_header().page_spin().set_value(page_number);
```

Comment:
I should probably only trigger the `set_value` if the page number is actually different from the actual value
+ use `block_signal`/`unblock_signal`  on the `page_spin` so that the function to change the page isn't called twice (thanks @Clemapfel for the idea)

Reply:
Seems like doing this is harder than it seems because [glib](https://gtk-rs.org/gtk-rs-core/git/docs/glib/index.html)::[signal](https://gtk-rs.org/gtk-rs-core/git/docs/glib/signal/index.html)::[SignalHandlerId](https://gtk-rs.org/gtk-rs-core/git/docs/glib/signal/struct.SignalHandlerId.html#) does not implement Clone, hence it's difficult to save the value and use it in another function to block/unblock things

Reply:
https://github.com/gtk-rs/gtk4-rs/issues/1817